TST: New failures in "numpy-dev" job?

I don't see how these are related to the `einsum` thingy that is known. Example: https://travis-ci.org/astropy/astropy/jobs/342443502

```
_________________________________ test__str__ __________________________________
        # non-scalar array angles
        arrangle = Angle(['10.2345d', '-20d'])
>       strarrangle = arrangle.__str__()
astropy/coordinates/tests/test_angles.py:861: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = 10.2345
>   return np.array2string(self, formatter={'all': lambda x: x.to_string()})
E   AttributeError: 'numpy.float64' object has no attribute 'to_string'
astropy/coordinates/angles.py:433: AttributeError
```

```
_______________________________ test_repr_latex ________________________________
        # and array angles
        arrangle = Angle([1, 2.1], u.deg)
>       rlarrangle = arrangle._repr_latex_()
astropy/coordinates/tests/test_angles.py:880: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = 1.0
>   return np.array2string(self, formatter={'all': lambda x: x.to_string(format='latex')})
E   AttributeError: 'numpy.float64' object has no attribute 'to_string'
astropy/coordinates/angles.py:441: AttributeError
```

```
______________________ TestTableFunctions.test_mask_array ______________________
        expect = r"\[\('NGC1002',\s+12.3[0-9]*\) \(\'NGC1003\',\s+15.[0-9]+\)\]"
>       assert re.match(expect, str(hdu.data))
E       assert None
E        +  where None = <function match at 0x7f9c2068fe18>("\\[\\('NGC1002',\\s+12.3[0-9]*\\) \\(\\'NGC1003\\',\\s+15.[0-9]+\\)\\]", "[(b'NGC1002', 12.3) (b'NGC1003', 15.2)]")
E        +    where <function match at 0x7f9c2068fe18> = re.match
E        +    and   "[(b'NGC1002', 12.3) (b'NGC1003', 15.2)]" = str(FITS_rec([(b'NGC1002', 12.3), (b'NGC1003', 15.2)],\n         dtype=(numpy.record, [('target', 'S20'), ('V_mag', '>f4')])))
E        +      where FITS_rec([(b'NGC1002', 12.3), (b'NGC1003', 15.2)],\n         dtype=(numpy.record, [('target', 'S20'), ('V_mag', '>f4')])) = <astropy.io.fits.hdu.table.BinTableHDU object at 0x7f9bb7bf6438>.data
astropy/io/fits/tests/test_table.py:860: AssertionError
```

```
_____________________ [doctest] docs/coordinates/index.rst _____________________
248 :func:`astropy.coordinates.EarthLocation.get_site_names`.
249 
250 For arbitrary Earth addresses (e.g., not observatory sites), use the
251 `~astropy.coordinates.EarthLocation.of_address` classmethod. Any address passed
252 to this function uses Google maps to retrieve the latitude and longitude and can
253 also (optionally) query Google maps to get the height of the location. As with
254 Google maps, this works with fully specified addresses, location names, city
255 names, and etc.::
256 
257     >>> EarthLocation.of_address('1002 Holy Grail Court, St. Louis, MO')  # doctest: +REMOTE_DATA +FLOAT_CMP
Expected:
    <EarthLocation (-26726.98216371, -4997009.8604809, 3950271.16507911) m>
Got:
    <EarthLocation (-26724.93406367, -4997006.87829221, 3950274.92599205) m>
/tmp/astropy-test-eem6lc45/lib/python3.6/site-packages/docs/coordinates/index.rst:257: DocTestFailure
```

```
_________________ [doctest] docs/io/fits/usage/unfamiliar.rst __________________
020 same, i.e. the data is in the ``.data`` attribute and the ``field()`` method
021 is used to refer to the columns and returns a numpy array. When reading the
022 table, Astropy will automatically detect what kind of table it is.
023 
024 ::
025 
026     >>> from astropy.io import fits
027     >>> filename = fits.util.get_testdata_filepath('ascii.fits')
028     >>> hdul = fits.open(filename)
029     >>> hdul[1].data[:1]  # doctest: +FLOAT_CMP
Expected:
    FITS_rec([(10.123, 37)],
             dtype=(numpy.record, {'names':['a','b'], 'formats':['S10','S5'], 'offsets':[0,11], 'itemsize':16}))
Got:
    FITS_rec([(b'.10123E+02', b'   37')],
             dtype=(numpy.record, {'names':['a','b'], 'formats':['S10','S5'], 'offsets':[0,11], 'itemsize':16}))
/tmp/astropy-test-eem6lc45/lib/python3.6/site-packages/docs/io/fits/usage/unfamiliar.rst:29: DocTestFailure
```

```>=================================== FAILURES ===================================
>_________________________________ test__str__ __________________________________
>   def test__str__():
>        """
>        Check the __str__ method used in printing the Angle
        """
    
        # scalar angle
        scangle = Angle('10.2345d')
        strscangle = scangle.__str__()
        assert strscangle == '10d14m04.2s'
    
        # non-scalar array angles
        arrangle = Angle(['10.2345d', '-20d'])
>       strarrangle = arrangle.__str__()
astropy/coordinates/tests/test_angles.py:864: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/coordinates/angles.py:437: in __str__
    return np.array2string(self, formatter={'all': lambda x: x.to_string()})
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:629: in array2string
    return _array2string(a, options, separator, prefix)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:426: in wrapper
    return f(self, *args, **kwargs)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:460: in _array2string
    summary_insert, options['legacy'])
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:757: in _formatArray
    curr_width=line_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:711: in recurser
    word = recurser(index + (-i,), next_hanging_indent, next_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:665: in recurser
    return format_function(a[index])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = 10.2345
>   return np.array2string(self, formatter={'all': lambda x: x.to_string()})
E   AttributeError: 'numpy.float64' object has no attribute 'to_string'
astropy/coordinates/angles.py:437: AttributeError
_______________________________ test_repr_latex ________________________________
    def test_repr_latex():
        """
        Check the _repr_latex_ method, used primarily by IPython notebooks
        """
    
        # try with both scalar
        scangle = Angle(2.1, u.deg)
        rlscangle = scangle._repr_latex_()
    
        # and array angles
        arrangle = Angle([1, 2.1], u.deg)
>       rlarrangle = arrangle._repr_latex_()
astropy/coordinates/tests/test_angles.py:883: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/coordinates/angles.py:445: in _repr_latex_
    return np.array2string(self, formatter={'all': lambda x: x.to_string(format='latex')})
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:629: in array2string
    return _array2string(a, options, separator, prefix)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:426: in wrapper
    return f(self, *args, **kwargs)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:460: in _array2string
    summary_insert, options['legacy'])
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:757: in _formatArray
    curr_width=line_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:711: in recurser
    word = recurser(index + (-i,), next_hanging_indent, next_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:665: in recurser
    return format_function(a[index])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = 1.0
>   return np.array2string(self, formatter={'all': lambda x: x.to_string(format='latex')})
E   AttributeError: 'numpy.float64' object has no attribute 'to_string'
astropy/coordinates/angles.py:445: AttributeError
_________________________________ test__str__ __________________________________
    def test__str__():
        """
        Check the __str__ method used in printing the Angle
        """
    
        # scalar angle
        scangle = Angle('10.2345d')
        strscangle = scangle.__str__()
        assert strscangle == '10d14m04.2s'
    
        # non-scalar array angles
        arrangle = Angle(['10.2345d', '-20d'])
>       strarrangle = arrangle.__str__()
astropy/coordinates/tests/test_angles.py:864: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/coordinates/angles.py:437: in __str__
    return np.array2string(self, formatter={'all': lambda x: x.to_string()})
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:629: in array2string
    return _array2string(a, options, separator, prefix)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:426: in wrapper
    return f(self, *args, **kwargs)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:460: in _array2string
    summary_insert, options['legacy'])
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:757: in _formatArray
    curr_width=line_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:711: in recurser
    word = recurser(index + (-i,), next_hanging_indent, next_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:665: in recurser
    return format_function(a[index])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = 10.2345
>   return np.array2string(self, formatter={'all': lambda x: x.to_string()})
E   AttributeError: 'numpy.float64' object has no attribute 'to_string'
astropy/coordinates/angles.py:437: AttributeError
_______________________________ test_repr_latex ________________________________
    def test_repr_latex():
        """
        Check the _repr_latex_ method, used primarily by IPython notebooks
        """
    
        # try with both scalar
        scangle = Angle(2.1, u.deg)
        rlscangle = scangle._repr_latex_()
    
        # and array angles
        arrangle = Angle([1, 2.1], u.deg)
>       rlarrangle = arrangle._repr_latex_()
astropy/coordinates/tests/test_angles.py:883: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/coordinates/angles.py:445: in _repr_latex_
    return np.array2string(self, formatter={'all': lambda x: x.to_string(format='latex')})
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:629: in array2string
    return _array2string(a, options, separator, prefix)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:426: in wrapper
    return f(self, *args, **kwargs)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:460: in _array2string
    summary_insert, options['legacy'])
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:757: in _formatArray
    curr_width=line_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:711: in recurser
    word = recurser(index + (-i,), next_hanging_indent, next_width)
/usr/lib64/python2.7/site-packages/numpy/core/arrayprint.py:665: in recurser
    return format_function(a[index])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = 1.0
>   return np.array2string(self, formatter={'all': lambda x: x.to_string(format='latex')})
E   AttributeError: 'numpy.float64' object has no attribute 'to_string'
astropy/coordinates/angles.py:445: AttributeError```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -786,6 +786,8 @@ astropy.coordinates
 
 - Add a workaround for a bug in the einsum function in Numpy 1.14.0. [#7187]
 
+- Fix problems with printing ``Angle`` instances under numpy 1.14.1. [#7234]
+
 astropy.cosmology
 ^^^^^^^^^^^^^^^^^
 

--- a/astropy/coordinates/angles.py
+++ b/astropy/coordinates/angles.py
@@ -14,6 +14,7 @@ import numpy as np
 from . import angle_utilities as util
 from .. import units as u
 from ..utils import isiterable
+from ..utils.compat import NUMPY_LT_1_14_1
 
 __all__ = ['Angle', 'Latitude', 'Longitude']
 
@@ -426,19 +427,26 @@ class Angle(u.SpecificTypeQuantity):
             ok &= np.all(self < Angle(upper))
         return bool(ok)
 
-    def __str__(self):
+    def _str_helper(self, format=None):
         if self.isscalar:
-            return str(self.to_string())
+            return self.to_string(format=format)
+
+        if NUMPY_LT_1_14_1:
+            def formatter(x):
+                return x.to_string(format=format)
         else:
-            return np.array2string(self, formatter={'all': lambda x: x.to_string()})
+            # In numpy 1.14.1, array2print formatters get passed plain numpy scalars instead
+            # of subclass array scalars, so we need to recreate an array scalar.
+            def formatter(x):
+                return self._new_view(x).to_string(format=format)
+
+        return np.array2string(self, formatter={'all': formatter})
+
+    def __str__(self):
+        return self._str_helper()
 
     def _repr_latex_(self):
-        if self.isscalar:
-            return self.to_string(format='latex')
-        else:
-            # Need to do a magic incantation to convert to str.  Regular str
-            # or array2string causes all backslashes to get doubled.
-            return np.array2string(self, formatter={'all': lambda x: x.to_string(format='latex')})
+        return self._str_helper(format='latex')
 
 
 def _no_angle_subclass(obj):

--- a/astropy/io/fits/tests/test_table.py
+++ b/astropy/io/fits/tests/test_table.py
@@ -18,6 +18,7 @@ except ImportError:
 
 from ....io import fits
 from ....tests.helper import catch_warnings, ignore_warnings
+from ....utils.compat import NUMPY_LT_1_14_1
 from ....utils.exceptions import AstropyDeprecationWarning
 
 from ..column import Delayed, NUMPY2FITS
@@ -844,6 +845,8 @@ class TestTableFunctions(FitsTestCase):
             assert header['TNULL2'] == 'b'
             assert header['TNULL3'] == 2.3
 
+    @pytest.mark.xfail(not NUMPY_LT_1_14_1,
+        reason="See https://github.com/astropy/astropy/issues/7214")
     def test_mask_array(self):
         t = fits.open(self.data('table.fits'))
         tbdata = t[1].data

--- a/astropy/utils/compat/numpycompat.py
+++ b/astropy/utils/compat/numpycompat.py
@@ -7,7 +7,8 @@ from ...utils import minversion
 
 
 __all__ = ['NUMPY_LT_1_10_4', 'NUMPY_LT_1_11', 'NUMPY_LT_1_11_2',
-           'NUMPY_LT_1_12', 'NUMPY_LT_1_13', 'NUMPY_LT_1_14']
+           'NUMPY_LT_1_12', 'NUMPY_LT_1_13', 'NUMPY_LT_1_14',
+           'NUMPY_LT_1_14_1']
 
 # TODO: It might also be nice to have aliases to these named for specific
 # features/bugs we're checking for (ex:
@@ -17,4 +18,5 @@ NUMPY_LT_1_11 = not minversion('numpy', '1.11.0')
 NUMPY_LT_1_11_2 = not minversion('numpy', '1.11.2')
 NUMPY_LT_1_12 = not minversion('numpy', '1.12')
 NUMPY_LT_1_13 = not minversion('numpy', '1.13')
-NUMPY_LT_1_14 = not minversion('numpy', '1.14dev')
+NUMPY_LT_1_14 = not minversion('numpy', '1.14')
+NUMPY_LT_1_14_1 = not minversion('numpy', '1.14.1')

--- a/docs/io/fits/usage/unfamiliar.rst
+++ b/docs/io/fits/usage/unfamiliar.rst
@@ -27,7 +27,12 @@ table, Astropy will automatically detect what kind of table it is.
     >>> filename = fits.util.get_testdata_filepath('ascii.fits')
     >>> hdul = fits.open(filename)
 
-.. doctest-requires:: numpy<=1.14.1
+..
+    This needs to be skipped for numpy-1.14.1. It should be removed when 1.14.2
+    is available. See https://github.com/astropy/astropy/issues/7214.
+
+.. doctest-skip::
+
     >>> hdul[1].data[:1]  # doctest: +FLOAT_CMP
     FITS_rec([(10.123, 37)],
              dtype=(numpy.record, {'names':['a','b'], 'formats':['S10','S5'], 'offsets':[0,11], 'itemsize':16}))
@@ -95,7 +100,12 @@ The other difference is the need to specify the table type when using the
     >>> col3 = fits.Column(name='t1', format='I', array=[91, 92, 93], ascii=True)
     >>> hdu = fits.TableHDU.from_columns([col1, col2, col3])
 
-.. doctest-requires:: numpy<=1.14.1
+..
+    This needs to be skipped for numpy-1.14.1. It should be removed when 1.14.2
+    is available. See https://github.com/astropy/astropy/issues/7214.
+
+.. doctest-skip::
+
     >>> hdu.data
     FITS_rec([('abc', 11.0, 91), ('def', 12.0, 92), ('', 0.0, 93)],
              dtype=(numpy.record, [('abc', 'S3'), ('def', 'S15'), ('t1', 'S10')]))

--- a/docs/io/fits/usage/unfamiliar.rst
+++ b/docs/io/fits/usage/unfamiliar.rst
@@ -26,6 +26,8 @@ table, Astropy will automatically detect what kind of table it is.
     >>> from astropy.io import fits
     >>> filename = fits.util.get_testdata_filepath('ascii.fits')
     >>> hdul = fits.open(filename)
+
+.. doctest-requires:: numpy<=1.14.1
     >>> hdul[1].data[:1]  # doctest: +FLOAT_CMP
     FITS_rec([(10.123, 37)],
              dtype=(numpy.record, {'names':['a','b'], 'formats':['S10','S5'], 'offsets':[0,11], 'itemsize':16}))
@@ -92,6 +94,8 @@ The other difference is the need to specify the table type when using the
     ...                    bzero=0.6, ascii=True)
     >>> col3 = fits.Column(name='t1', format='I', array=[91, 92, 93], ascii=True)
     >>> hdu = fits.TableHDU.from_columns([col1, col2, col3])
+
+.. doctest-requires:: numpy<=1.14.1
     >>> hdu.data
     FITS_rec([('abc', 11.0, 91), ('def', 12.0, 92), ('', 0.0, 93)],
              dtype=(numpy.record, [('abc', 'S3'), ('def', 'S15'), ('t1', 'S10')]))

